src/components/TrackPain/Canvas.js

```diff
@@ -97,7 +92,8 @@ class Canvas extends Component {
    * @return {void} body part name
    */
   getBodyPartTerminology(x, y) {
-    const { color, front } = this.state;
+    const { front } = this.state;
+    front === null && this.setState({ front: false });
```

Comment:
If we set the initial state to false, this line is not necessary. For example:
```js
this.state = {
  front: false,
 ...
};
...
front === null && this.setState({ front: false }); // <== not necessary. We can delete it.
```

Reply:
This line is necessary because in the case of new entries the attribute "front" will be null, at least you click the button to switch the human body silhouette. It is not good to send this attribute null to the database, Firebase will not create it. You can try this: 

1. Go to tracking Pain.
2. Click in add new entry.
3. Without making any switch of the human body silhouette complete and save the entry.
4. Back in tracking pain edit the entry you just saved.
5. Draw something else in the canvas, again, without making any switch of the human body silhouette, and try to save it again. You will see "Error: Reference.update failed: First argument contains undefined in property ...". That is because the entry doesn't have the front attribute in Firebase.